test/ChainSpecificDeployment.js

```diff
@@ -0,0 +1,136 @@
+'use strict';
+
+const { ethers } = require('hardhat');
+const { expect } = require('chai');
+const { deployAll } = require('../scripts/deploy');
+
+describe('Chain-Specific Token Deployment', () => {
+    let wallet;
+
+    before(async () => {
+        [wallet] = await ethers.getSigners();
+    });
+
+    it('should deploy HyperliquidInterchainToken for Hyperliquid chain', async () => {
+        console.log('\n=== Testing Hyperliquid Chain Deployment ===');
+
+        const deployment = await deployAll(
+            wallet,
+            'hyperliquid',
+            [],
+            'HyperliquidInterchainTokenService',
+            'HyperliquidInterchainTokenServiceFactory',
+        );
+
+        console.log('Deployed contracts:');
```

Comment:
too much superfluous code in tests, please clean it up. harder to review, easy to miss the critical changes